src/map_extras.cpp

```diff
@@ -186,12 +189,13 @@ void mx_helicopter( map &m, const tripoint &abs_sub )
                 }
                 break;
             case 6: // Just pilots
-                for( auto p : wreckage->get_parts( VPFLAG_CONTROLS, false, true ) ) {
+                for( const vpart_reference vp : wreckage->get_parts_including_broken( VPFLAG_CONTROLS ) ) {
+                    const vehicle_part *const p = &vp.vehicle().parts[vp.part_index()];
```

Comment:
Any particular this is a pointer instead of a reference?

Reply:
The current master already has this pointer variable. I just changed how it's calculated. This is going to be changed later.

Reply:
Yeah, it's just way more obvious now that's there's places where it's taking a pointer and then immediately converting it back.  If you're already planning on fixing it another PR then disregard. 
 Thanks.  Also thanks for cleaning up those `auto`s.